src/_zkapauthorizer/recover.py

```diff
@@ -171,22 +170,23 @@ async def canned_downloader(set_state: SetState) -> BinaryIO:
 noop_downloader = make_canned_downloader(statements_to_snapshot(iter([])))
 
 
-def statements_from_snapshot(data: BinaryIO) -> Iterator[str]:
+def statements_from_snapshot(get_snapshot: DataProvider) -> Iterator[str]:
```

Comment:
Nah.  It just seemed like a reasonable type?  I didn't want to pass around open file objects (because then how are you sure they'll be closed) or FilePaths (because the tests use BytesIO all over the place) so "a function to call to get a file object" seemed like the right thing.  It lets the user do open/close management and it lets the tests keep avoiding the filesystem.